### PR TITLE
Add support for PHP

### PR DESCRIPTION
**Pull request by pawan-privado**
This PR checks in the processor for `PHP` as part of support for `PHP` language.

### CODE REVIEW

**Comment by khemrajrathore** [comment]
Draft changes are good, let's also integrate PrivadoTagger

**Comment by pawan-privado** [comment]
Thanks, already on it!

File: src/main/scala/ai/privado/languageEngine/php/tagger/PrivadoTagger.scala [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+  override def runTagger(
+    rules: RuleCache,
+    taggerCache: TaggerCache,
+    privadoInputConfig: PrivadoInput,
+    dataFlowCache: DataFlowCache
+  ): Traversal[Tag] = {
+    logger.info("Beginning tagging")
+
+    new LiteralTagger(cpg, rules).createAndApply()
+    new RegularSinkTagger(cpg, rules).createAndApply()
```
Comment (khemrajrathore) [inline src/main/scala/ai/privado/languageEngine/php/tagger/PrivadoTagger.scala:49]: Place SinkTagger After IdentifierTagger, as we first complete source detection and then sink detection.

**Comment by pawan-privado** [comment]
Update: we are now able to scan `PHP` repos successfully.

**Review by tuxology — CHANGES_REQUESTED** [review]
This needs just a little bit more work 🤝 

While building and generating CPG via via `./privado-core -ic <path to rules> scan <path to source>` we faced the following issue:

```
2024-02-29 00:15:00.122 ERROR ai.privado.cache.Environment$ Environment.scala:50: Error fetching version for privado-core
Privado CLI Version: Not Detected
Privado Core Version: Not Detected
Privado Language Engine Version: 2.0.277
Thu Feb 29 00:15:00 UTC 2024 - Language detection done in                       - 0 ms - 00h:00m:00s:00ms
Thu Feb 29 00:15:00 UTC 2024 - Detected language 'PHP'
Privado Main Version: Not Detected
Thu Feb 29 00:15:02 UTC 2024 - Configuration parsed...
Thu Feb 29 00:15:02 UTC 2024 - Processing source code using php engine
2024-02-29 00:15:02.669 ERROR io.joern.php2cpg.parser.PhpParser$ PhpParser.scala:103: Invalid path for PhpParserBin: /home/Suchakra/suchakra-testing/privado-core-pawan/privado-core/target/universal/stage/lib/io.joern./php2cpg/bin/php-parser/php-parser.php
2024-02-29 00:15:02.698 ERROR io.joern.php2cpg.Php2Cpg Php2Cpg.scala:75: Skipping AST creation as php/php-parser could not be executed.
- Could not initialize PhpParser
2024-02-29 00:15:02.699 ERROR ai.privado.languageEngine.php.processor.PhpProcessor BaseProcessor.scala:77: Error while parsing the source code!
2024-02-29 00:15:02.700 ERROR ai.privado.entrypoint.Main$ Main.scala:64: Error in scanning, skipping auth flow : Error while parsing the source code
```

Note the incorrect path for php-parser containing `... /io.joern./php2cpg/bin/php-parser ...` This is mostly due to the fact the PhpParserBin is getting incorrect path since it builds path relative to the joern directory when it is executed from Joern: https://github.com/joernio/joern/blob/master/joern-cli/frontends/php2cpg/src/main/scala/io/joern/php2cpg/parser/PhpParser.scala#L80 The way Joern manages these paths is by vendoring `PHP-Parse`. See https://github.com/joernio/joern/blob/master/joern-cli/frontends/php2cpg/build.sbt#L23

To fix this issues, we have two alternatives:
 1. Either vendor `PHP-Parser` same way as Joern and then pass PHP_PARSE_BIN or custom path via `Config` to the parser
 2. Figure out how to fix this in upstream Joern via changing the path relative the parent library that vendors Joern itself. 

I am inclined to use option 1 so we have better control over Parser and we can try to build/switch parsers faster if needed later on.

**Pull request merged**